### PR TITLE
[3718]external storage permission xml

### PR DESCRIPTION
**Pull request by TGazica**
### 🎯 Goal

Add `WRITE_EXTERNAL_STORAGE` permission checks and prompts for API 28 and under.

### 🛠 Implementation details

Change the default implementation of the download handler inside `MessageListViewModel.bindView`.

### 🧪 Testing

1. Run a fresh install of develop on a device with API 28 or under.
2. Try to download a file (don't open attachments or grant any permissions)
3. Build the app from this branch and repeat step 1 and 2

#### General
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [x] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### 🎉 GIF

giphy

### CODE REVIEW

File: stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/ChatFragment.kt [hunk 1]
```diff
@@ -75,9 +76,13 @@ class ChatFragment : Fragment() {
         _binding = null
     }
 
+    private val downloadHandler: DownloadPermissionHandler = DownloadPermissionHandler()
```
Comment (filbabic) [inline stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/ChatFragment.kt:79]: comments

File: stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/ChatFragment.kt [hunk 2]
```diff
@@ -61,6 +62,13 @@ class ChatFragment : Fragment() {
     private var _binding: FragmentChatBinding? = null
     private val binding get() = _binding!!
 
+    /**
+     * [DownloadPermissionHandler] instance tasked with requesting [Manifest.permission.WRITE_EXTERNAL_STORAGE]
+     * permission. By default will ask for permission and if it is granted will download the file.
+     */
+    private val downloadHandler: DownloadPermissionHandler = DownloadPermissionHandler()
```
Comment (filbabic) [inline stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/ChatFragment.kt:69]: can we make this a lazy property

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/MessageListFragment.kt [hunk 3]
```diff
@@ -92,10 +92,12 @@ public open class MessageListFragment : Fragment() {
...
 
-    override fun onAttach(context: Context) {
-        super.onAttach(context)
-        backPressListener = findListener()
-    }
+    /**
+     * [DownloadPermissionHandler] instance tasked with requesting [Manifest.permission.WRITE_EXTERNAL_STORAGE]
+     * permission. By default will ask for permission and if it is granted will download the file.
+     */
+    private val downloadHandler: DownloadPermissionHandler = DownloadPermissionHandler()
```
Comment (filbabic) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/MessageListFragment.kt:99]: same here - lazy if possible

File: stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/ChatFragment.kt (hunk 2 above)
Comment (TGazica) [reply stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/ChatFragment.kt:69]: Conflicts with the activity lifecycle stuff

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/MessageListFragment.kt [hunk 4]
```diff
@@ -92,10 +92,12 @@ public open class MessageListFragment : Fragment() {
     private var _binding: StreamUiFragmentMessageListBinding? = null
     protected val binding: StreamUiFragmentMessageListBinding get() = _binding!!
 
-    override fun onAttach(context: Context) {
```
Comment (bychkovdmitry) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/MessageListFragment.kt:95]: Why is this code removed?

Comment (TGazica) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/MessageListFragment.kt:95]: i have no idea to be honest, maybe i did by accident, but dont remember doing it, will revert it.

**Review by bychkovdmitry — APPROVED** [review]
Works well on API 21, API 26, API 29

**Pull request merged**